Gemfile

```diff
@@ -1,3 +1,7 @@
 source "https://rubygems.org"
 
 gemspec
+
+unless ENV["NO_STEEP"]
```

Comment:
Using `group` would be better for this but I'm not sure if there is any specific reason it doesn't work...

Reply:
No, that should work just fine. It didn't occur to me because I've only ever used them coupled to rails environments -.-

Reply:
well.. Actually I can't seem to make it work, but only in github actions on the 2.7 container. Setting the `BUNDLE_WITHOUT=typing` environment variable suffices in all of the other slices of the matrix [here](https://github.com/nevinera/environment_helpers/actions/runs/6925239654/job/18835599187) to skip installing steep, but on 2.7 I still get the error message

```
Because steep >= 1.6.0.pre.1 depends on Ruby >= 3.0.0
  and Gemfile depends on steep ~> 1.6.0,
  Ruby >= 3.0.0 is required.
```

It's quite irritating, because that works fine locally on the same combination of ruby/bundler versions. It the action it also seems to ignore being supplied `--without` directly (despite it being deprecated) and my installing them via `bundle config set --local without 'typing' && bundle install`.

If what I'm doing wrong is obvious to you, please de-frustrate me, but otherwise I think I'll just leave it - I'm actually just supplying a _different_ environment variable and using that to control the group-selection, so in the end it's not really improving anything, just being a little clearer about intentions.